[TSK-388] ChatPage and SendMessageToConversation feature

alebg opened the pull request:
Implementation following the latest conventions

alebg committed:
fix: symbol wiring for research context trpc router

alebg committed:
feat: implemented SendMessageToConversation backend primary side

alebg commented:
@maany Not done yet, just opening this PR to be reviewable already. Implementing in this order: 

1. Primary side back end minus usecase (done)
2. Page (to-do: need coordination with whoever is doing the chat components in rage-ui-kit)
3. Secondary side back end (blocked by KP tasks)
4. Usecase (blocked by 3)

alebg committed:
feat: wired dummy pages to chat

- try: control for whenever the current client doesn't have the research context or conversation they're navigating to
- fix: IDs in backend turned to numbers (from strings)

maany commented on src/app/[rc_id]/conversations/[conv_id]/page.tsx:
```diff
@@ -0,0 +1,87 @@
+  const listResearchContextsDTO = await researchContextGateway.list()
+  if (!listResearchContextsDTO.success) {
+    // TODO: discuss what to do here
+    redirect("/");
```
redirect to nextjs error page

maany commented on src/app/[rc_id]/conversations/[conv_id]/page.tsx:
```diff
@@ -0,0 +1,87 @@
+  const listConversationsDTO = await conversationGateway.listConversations(researchContextID);
+  if (!listConversationsDTO.success) {
+    // TODO: discuss what to do here
+    redirect(`/${researchContextID}/conversations`);
```
why not to /${researchContextID}?

alebg committed:
feat: scaffolding and wiring for sendMessageToConversation feature

- Client-side usecase wired (with the correct presenter, logger, and client side gateways)
- Signal wired
- Client-side gateways do TRPC calls to the server-side gateways; TRPC routers wired
- Cleanup of function signatures and types
- Implemented page and wired it to the UI kit's chat component
- TODO: OpenAI server-side gateways missing implementation (prepare message context, and send message); new KP client missing, which will imply some small refactoring of the server-side kernel gateway function send message; adapt page and client component to changes

alebg committed:
update: ui kit version

alebg commented:
Closed in favor of  #52
